2_process/src/do_lake_toha_tasks.R

```diff
@@ -53,13 +51,37 @@ create_lake_tasks <- function(task_df_fn, log_folder){
     } 
   )
   
-  # ---- combine into a task plan ---- #
+  ##-- Create the task plan --##
   
-  gif_task_plan <- scipiper::create_task_plan(
+  task_plan <- create_task_plan(
     task_names=tasks$site_id,
     task_steps=list(split_morphometry,
-      calculate_model_toha),
+                    calculate_model_toha),
     add_complete=FALSE,
-    final_steps=c(toha_step_name),
-    ind_dir='2_process/log')
+    final_steps=c(toha_step_name))
+  
+  ##-- Create the task remakefile --##
+  
+  task_makefile <- sprintf('2_%s_lake_tasks.yml', model_prefix)
+  create_task_makefile( 
+    task_plan=task_plan,
+    makefile=task_makefile,
+    include='remake.yml',
+    sources=c(...),
+    packages=c("purrr", "dplyr", "mda.lakes", "feather", "rLakeAnalyzer"),
+    final_targets = final_target,
+    as_promises = TRUE,
+    tickquote_combinee_objects = TRUE
+  )
+  
+  ##-- Build the tasks --##
+  
+  loop_tasks(task_plan = task_plan,
+             task_makefile = task_makefile,
+             num_tries = 1, 
+             n_cores = n_cores)
+  
+  # Now return the file name of the final target
```

Comment:
when I'm not checking the task table yaml into git, I've been deleting it after use as a clean up (I have also been deleting the combiner promise target before returning). This is a preference thing, so I don't think you need to do the same unless you would prefer to. 

But the reason I `scdel()` the promise target is because you may accidentally delete the file that is built as a side-effect of the promise target, or alter it outside of the task table for some reason. If the file was deleted and you call this whole function to rebuild it, I _think_ remake will hit that final promise target and _not_ rebuild it because in the remake database, that promise target is current. This can be avoided by getting rid of the promise after you use it...but this is kind of an edge case issue and you can use your preference on whether or not it makes sense to do the same here (this is the pattern we have in `slim-data-release`)

Reply:
I think this would be useful because it would help my directory stay cleaner. There are lots of these tasks ymls that get created in the process and it would be nice to delete once they are not needed.

Ahhh good point on the `_promise` gotcha. I can add that pattern in too. I don't see why I would be altering it outside of the task_table, but seems simple enough to implement just in case.